engine/src/test/java/org/camunda/bpm/engine/test/concurrency/CompetingExternalTaskFetchingTest.java

```diff
@@ -113,6 +119,11 @@ public void testCompetingExternalTaskFetching() {
     thread2.proceedAndWaitTillDone();
     assertEquals(0, thread2.fetchedTasks.size());
     // but does not fail with an OptimisticLockingException
-    assertNull(thread2.exception);
+    if (testRule.isOptimisticLockingExceptionSuppressible()) {
+      assertNull(thread2.exception);
+    } else {
+      // CockroachDb
+      assertTrue(thread2.exception instanceof CrdbTransactionRetryException);
+    }
```

Comment:
The command fails since by default we have the `commandRetries` property set to 0. So even if a retryable command fails, it  will not be retried, but the exception will be re-thrown to the caller. In this case, the exception is expected.

We can also discuss this further, if I didn't understand your question.